### PR TITLE
Test and Score: Update documentation and update the file name (and class name)

### PR DESCRIPTION
**Pull request by PrimozGodec**
##### Issue

Documentation for Test and Score was out of date - old images, info about some new features missing.
Test and Score still have an old name Test Learners.

##### Description of changes
Updated documentation for Test and Score
File and class name changed from Test Learners to Test and Score

##### Includes
- [X] Documentation

### CODE REVIEW

File: doc/visual-programming/source/widgets/evaluate/testandscore.md [hunk 1]
```diff
@@ -17,38 +17,42 @@ The widget tests learning algorithms. Different sampling schemes are available,
...
    - **Test on train data** uses the whole dataset for training and then for testing. This method practically always gives wrong results.
    - **Test on test data**: the above methods use the data from *Data* signal only. To input another dataset with testing examples (for instance from another file or some data selected in another widget), we select *Separate Test Data* signal in the communication channel and select Test on test data.
 2. For classification, *Target class* can be selected at the bottom of the widget. When *Target class* is (Average over classes), methods return scores that are weighted averages over all classes. For example, in case of the classifier with 3 classes, scores are computed for class 1 as a target class, class 2 as a target class, and class 3 as a target class. Those scores are averaged with weights based on the class size to retrieve the final score.
-3. Produce a report.
-4. The widget will compute a number of performance statistics, which can be shown or hidden by right-clicking the columns:
+3. The widget will compute a number of performance statistics, which can be shown or hidden by right-clicking the columns:
```
Comment (ajdapretnar) [inline doc/visual-programming/source/widgets/evaluate/testandscore.md]: I think the wording here is not quite ok (this is probably some of my old stuff). Would you mind changing this to:

> The widget will compute a number of performance statistics. A few are shown by default. To see others, right-click on the header and select the desired statistic.

Or something similar.

File: doc/visual-programming/source/widgets/evaluate/testandscore.md [hunk 2]
```diff
@@ -17,38 +17,42 @@ The widget tests learning algorithms. Different sampling schemes are available,
...
+        - Train time - cumulative time in seconds used for training models.
+        - Test time - cumulative time in seconds used for testing models.
    - Regression
-   ![](images/TestLearners-regression.png)
+   ![](images/TestAndScore-Regression.png)
       - [MSE](https://en.wikipedia.org/wiki/Mean_squared_error) measures the average of the squares of the errors or deviations (the difference between the estimator and what is estimated).
       - [RMSE](https://en.wikipedia.org/wiki/Root_mean_square) is the square root of the arithmetic mean of the squares of a set of numbers (a measure of imperfection of the fit of the estimator to the data)
       - [MAE](<https://en.wikipedia.org/wiki/Mean_absolute_error>) is used to measure how close forecasts or predictions are to eventual outcomes.
       - [R2](<https://en.wikipedia.org/wiki/Coefficient_of_determination>) is interpreted as the proportion of the variance in the dependent variable that is predictable from the independent variable.
```
Comment (ajdapretnar) [inline doc/visual-programming/source/widgets/evaluate/testandscore.md]: Could you add CVRMSE and train time and test time?

File: doc/visual-programming/source/widgets/evaluate/testandscore.md [hunk 3]
```diff
@@ -17,38 +17,42 @@ The widget tests learning algorithms. Different sampling schemes are available,
...
    - Classification
-   ![](images/TestLearners.png)
+   ![](images/TestAndScore-Classification.png)
         - [Area under ROC](http://gim.unmc.edu/dxtests/roc3.htm) is the area under the receiver-operating curve.
         - [Classification accuracy](https://en.wikipedia.org/wiki/Accuracy_and_precision) is the proportion of correctly classified examples.
         - [F-1](https://en.wikipedia.org/wiki/F1_score) is a weighted harmonic mean of precision and recall (see below).
         - [Precision](https://en.wikipedia.org/wiki/Precision_and_recall) is the proportion of true positives among instances classified as positive, e.g. the proportion of *Iris virginica* correctly identified as Iris virginica.
         - [Recall](https://en.wikipedia.org/wiki/Precision_and_recall) is the proportion of true positives among all positive instances in the data, e.g. the number of sick among all diagnosed as sick.
+        - [Specificity](https://en.wikipedia.org/wiki/Sensitivity_and_specificity) is the proportion of true negatives among all negative instances, e.g. the number of non-sick among all diagnosed as non-sick.
```
Comment (ajdapretnar) [inline doc/visual-programming/source/widgets/evaluate/testandscore.md]: Could you add LogLoss?

**Comment by ajdapretnar** [comment]
It would be nice to have an example here with the new model comparison, but if there is no immediate use case that we can think of, we can leave it as it is.

**Comment by ajdapretnar** [comment]
Btw, @PrimozGodec, I see you changed the outdated Test Learners wording in the docs. I fully support this, so perhaps it is time we change the name of the file, too? It would be a good OSS practice to name things for what they are. :D

**Comment by janezd** [comment]
This changes the widget's qualified name. If the change can be done without breaking the compatiblity (with "replaces") I strongly support it.

**Comment by PrimozGodec** [comment]
I will add the suggestions above. 

I think changing the name of the file in the documentation does not change anything. As far as I know widget match to the documentation via widget name and documentation title - the first line. Did I miss anything here? 
@janezd did you also think to change the name of the widget file?

**Comment by PrimozGodec** [comment]
> It would be nice to have an example here with the new model comparison, but if there is no immediate use case that we can think of, we can leave it as it is.

Since the current example already shows the models comparison I just added a paragraph with the description.

> Btw, @PrimozGodec, I see you changed the outdated Test Learners wording in the docs. I fully support this, so perhaps it is time we change the name of the file, too? It would be a good OSS practice to name things for what they are. :D

> This changes the widget's qualified name. If the change can be done without breaking the compatibility (with "replaces") I strongly support it.

It seems that "replaces" servers well. I changed the name and tested it with the saved workflow.

File: doc/visual-programming/source/widgets/evaluate/testandscore.md [hunk 4]
```diff
@@ -17,38 +17,48 @@ The widget tests learning algorithms. Different sampling schemes are available,
...
+      - [CVRMSE](https://en.wikipedia.org/wiki/Root-mean-square_deviation) is RMSE normalized by the mean value of actual values.
+      - Train time - cumulative time in seconds used for training models.
+      - Test time - cumulative time in seconds used for testing models.
+4. Choose the score for pairwise comparison of models and the region of practical equivalence (ROPE), in which differences are considered negligible.
+5. Pairwise comparison of models using the selected score (available only for cross-validation). The number in the table gives the probability that the model corresponding to the row is better than the model corresponding to the column. If negligible difference is enabled, the smaller number below shows the probability that the difference between the pair is negligible. The test is based on the Bayesian interpretation of the t-test; (paper: https://link.springer.com/article/10.1007/s10994-015-5486-z,16-305.html, shorter introduction: https://baycomp.readthedocs.io/en/latest/introduction.html).
```
Comment (ajdapretnar) [inline doc/visual-programming/source/widgets/evaluate/testandscore.md:52]: Sorry, missed this one. The first link doesn't work. Both should preferably be formatted as links. Currently, one has to copy-paste.

Comment (PrimozGodec) [reply doc/visual-programming/source/widgets/evaluate/testandscore.md:52]: I also didn't notice that. Those links were in the documentation before already. Will fix them.

Comment (ajdapretnar) [reply doc/visual-programming/source/widgets/evaluate/testandscore.md]: Oops. Might be my fault then. 😅 🙈

Comment (PrimozGodec) [reply doc/visual-programming/source/widgets/evaluate/testandscore.md]: Nope, not your fault. :)

**Pull request merged**